### ISSUE TITLE
Disabling SEO extension causes fatal error (^1.0)

### ISSUE THREAD

**Issue by iamdarkle** [v17development/flarum-seo#67]
Initially (after upgrading from beta 16 to stable 1.0) when I disabled the extension (investigating other bugs) I noticed that it directly caused a HTTP ERROR 500 so the console doesn't show any data and I didn't get any log in storage/logs about it. 

After removing the extension and reinstalling it, I disabled it and this time it did throw an error (debug mode enabled):

```
Exception thrown with message "V17Development\FlarumSeo\Extend::setTitle(..): Container was not yet initialized."

Stacktrace:
#60 Exception in /var/www/flarum/vendor/v17development/flarum-seo/src/Extend.php:196
#59 V17Development\FlarumSeo\Extend:throwError in /var/www/flarum/vendor/v17development/flarum-seo/src/Extend.php:36
#58 V17Development\FlarumSeo\Extend:setTitle in /var/www/flarum/vendor/v17development/flarum-blog/src/Controller/BlogOverviewController.php:40
#57 V17Development\FlarumBlog\Controller\BlogOverviewController:__invoke in /var/www/flarum/vendor/flarum/core/src/Frontend/Frontend.php:62
#56 Flarum\Frontend\Frontend:populate in /var/www/flarum/vendor/flarum/core/src/Frontend/Frontend.php:54
#55 Flarum\Frontend\Frontend:document in /var/www/flarum/vendor/flarum/core/src/Frontend/Controller.php:32
#54 Flarum\Frontend\Controller:handle in /var/www/flarum/vendor/flarum/core/src/Http/RouteHandlerFactory.php:41
#53 Flarum\Http\RouteHandlerFactory:Flarum\Http\{closure} in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/ExecuteRoute.php:27
#52 Flarum\Http\Middleware\ExecuteRoute:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#51 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/fof/terms/src/Middlewares/RegisterMiddleware.php:48
#50 FoF\Terms\Middlewares\RegisterMiddleware:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#49 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/fof/discussion-language/src/Middleware/AddLanguageFilter.php:32
#48 FoF\DiscussionLanguage\Middleware\AddLanguageFilter:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#47 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/fof/oauth/src/Middleware/ErrorHandler.php:46
#46 FoF\OAuth\Middleware\ErrorHandler:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#45 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/ContentTypeOptionsHeader.php:21
#44 Flarum\Http\Middleware\ContentTypeOptionsHeader:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#43 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/ReferrerPolicyHeader.php:30
#42 Flarum\Http\Middleware\ReferrerPolicyHeader:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#41 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/FlarumPromotionHeader.php:30
#40 Flarum\Http\Middleware\FlarumPromotionHeader:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#39 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/ShareErrorsFromSession.php:57
#38 Flarum\Http\Middleware\ShareErrorsFromSession:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#37 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/CheckCsrfToken.php:36
#36 Flarum\Http\Middleware\CheckCsrfToken:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#35 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/ResolveRoute.php:67
#34 Flarum\Http\Middleware\ResolveRoute:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#33 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/SetLocale.php:51
#32 Flarum\Http\Middleware\SetLocale:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#31 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/AuthenticateWithSession.php:31
#30 Flarum\Http\Middleware\AuthenticateWithSession:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#29 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/RememberFromCookie.php:52
#28 Flarum\Http\Middleware\RememberFromCookie:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#27 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/StartSession.php:61
#26 Flarum\Http\Middleware\StartSession:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#25 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/CollectGarbage.php:46
#24 Flarum\Http\Middleware\CollectGarbage:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#23 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/ParseJsonBody.php:28
#22 Flarum\Http\Middleware\ParseJsonBody:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#21 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/migratetoflarum/canonical/src/Middlewares/CanonicalRedirectMiddleware.php:49
#20 MigrateToFlarum\Canonical\Middlewares\CanonicalRedirectMiddleware:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#19 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/HandleErrors.php:57
#18 Flarum\Http\Middleware\HandleErrors:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#17 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/InjectActorReference.php:25
#16 Flarum\Http\Middleware\InjectActorReference:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#15 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php:84
#14 Laminas\Stratigility\MiddlewarePipe:process in /var/www/flarum/vendor/middlewares/request-handler/src/RequestHandler.php:84
#13 Middlewares\RequestHandler:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#12 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/middlewares/base-path-router/src/BasePathRouter.php:101
#11 Middlewares\BasePathRouter:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#10 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Middleware/OriginalMessages.php:42
#9 Laminas\Stratigility\Middleware\OriginalMessages:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#8 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/middlewares/base-path/src/BasePath.php:73
#7 Middlewares\BasePath:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#6 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/flarum/core/src/Http/Middleware/ProcessIp.php:24
#5 Flarum\Http\Middleware\ProcessIp:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/Next.php:61
#4 Laminas\Stratigility\Next:handle in /var/www/flarum/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php:84
#3 Laminas\Stratigility\MiddlewarePipe:process in /var/www/flarum/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php:73
#2 Laminas\Stratigility\MiddlewarePipe:handle in /var/www/flarum/vendor/laminas/laminas-httphandlerrunner/src/RequestHandlerRunner.php:96
#1 Laminas\HttpHandlerRunner\RequestHandlerRunner:run in /var/www/flarum/vendor/flarum/core/src/Http/Server.php:44
#0 Flarum\Http\Server:listen in /var/www/flarum/public/index.php:26

```

Server details:
```
Operating System: CentOS Linux 8
Kernel: Linux 4.18.0-240.15.1.el8_3.x86_64
Flarum core 0.1.0-beta.16
nginx version: nginx/1.18.0
10.3.28-MariaDB
PHP version: 8.0.3
Loaded extensions: Core, date, libxml, openssl, pcre, zlib, filter, hash, pcntl, readline, Reflection, SPL, session, standard, bz2, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, intl, json, exif, mysqlnd, PDO, Phar, SimpleXML, sockets, sodium, sqlite3, tokenizer, xml, xmlwriter, xsl, mysqli, pdo_mysql, pdo_sqlite, xmlreader, Zend OPcache
```

**Comment by clarkwinkelmann** [comment]
This bug still exists with Flarum 1.4 and the latest version of Blog and SEO. These are the conditions:

If `v17development/flarum-blog` is enabled and `v17development/flarum-seo` is installed but disabled, trying to access any Blog page causes a 500 error.

That's because Blog merely checks whether the PHP classes for SEO exist, which remains true when the extension is disabled.

https://github.com/v17development/flarum-blog/blob/e09de191f5aba0808b05e859eec14c147649537e/src/Controller/BlogItemController.php#L43-L88
https://github.com/v17development/flarum-blog/blob/e09de191f5aba0808b05e859eec14c147649537e/src/Controller/BlogOverviewController.php#L28-L41

This could be patched either in Blog extension by checking whether SEO is enabled before calling its extenders, or refactoring SEO so that calling its extenders whilst disabled just becomes a no-op.

**Comment by jaspervriends** [comment]
This bug will be fixed in version 2.0 that will be released soon, thanks for reporting 🚀
